aiven_mysql_migrate/dump_tools.py

```diff
@@ -143,33 +143,39 @@ def __init__(
         skip_column_stats: bool,
         *,
         dump_tool_name: MySQLMigrateTool = MySQLMigrateTool.mydumper,
+        temp_dir: Optional[Path] = None,
     ):
         super().__init__(source, target, databases, skip_column_stats, dump_tool_name=dump_tool_name)
-        self.temp_dir: Optional[tempfile.TemporaryDirectory] = None
+        self._base_temp_dir: Optional[Path] = temp_dir
+        self.temp_dir_path: Optional[Path] = None
+        self._temp_dir_obj: Optional[tempfile.TemporaryDirectory] = None
         self.temp_cnf_file: Optional[Path] = None
         self.temp_target_cnf_file: Optional[Path] = None
         self.dump_output_dir: Optional[Path] = None
 
     def setup(self) -> None:
         """Setup temporary resources for migration."""
 
-        if not self.temp_dir:
-            self.temp_dir = tempfile.TemporaryDirectory()  # pylint: disable=consider-using-with
+        if not self.temp_dir_path:
+            # pylint: disable=consider-using-with
+            self._temp_dir_obj = tempfile.TemporaryDirectory(dir=self._base_temp_dir, prefix="mydumper_")
+            self.temp_dir_path = Path(self._temp_dir_obj.name)
+            self._base_temp_dir = None  # No longer needed after setup
```

Comment:
This is annoyingly convoluted but actually I couldn't think of a more elegant alternative.